lib/earmark/block.ex

```diff
@@ -168,14 +168,34 @@ defmodule Earmark.Block do
   # in the second we combine adjacent items into lists. This is pass one
 
   defp _parse( [first = %Line.ListItem{type: type, initial_indent: initial_indent, content: content, bullet: bullet, lnb: lnb} | rest ], result, options) do
-    {spaced, list_lines, rest, _offset, indent_level} = read_list_lines(rest, opens_inline_code(first), initial_indent)
+    {lines_spaced, list_lines, rest, _offset, indent_level} = read_list_lines(rest, opens_inline_code(first), initial_indent)
 
-    spaced = (spaced || blank_line_in?(list_lines)) && peek(rest, Line.ListItem, type)
     lines = for line <- list_lines, do: indent_list_item_body(line, indent_level || 0)
     lines = [content | lines]
     {blocks, _, options1} = Parser.parse(lines, %{options | line: lnb}, true)
 
-    _parse([%Line.Blank{lnb: 0} | rest], [ %ListItem{type: type, blocks: blocks, spaced: spaced, bullet: bullet, lnb: lnb} | result ], options1)
+    # Determine whether the list item's direct content contains blank lines.
+    # (Blanks within a sublist don't count.)
+    blanks = blank_line_in?(list_lines) && !contains_list?(blocks)
```

Comment:
I have to add more tests to see if this is sufficient.  Don't know if it's possible that it contains direct content with blanks and a sublist.  It doesn't seem like it, but I'm not quite sure.